sonic-xcvrd/scripts/xcvrd

```diff
@@ -325,28 +349,130 @@ def del_port_sfp_dom_info_to_db(logical_port_name, int_tbl, dom_tbl):
             log_error("This functionality is currently not implemented for this platform")
             sys.exit(3)
 
-# Timer thread wrapper class to update dom info to DB periodically
+# wait for port config is done
+def wait_for_port_config_done():
+    # Connect to APPL_DB abd subscribe to PORT table notifications
+    appl_db = swsscommon.DBConnector(swsscommon.APPL_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_MSECS)
+
+    sel = swsscommon.Select()
+    sst = swsscommon.SubscriberStateTable(appl_db, swsscommon.APP_PORT_TABLE_NAME)
+    sel.addSelectable(sst)
+
+    # Make sure this daemon started after all port configured
+    while RUN:
+        (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
+        if state == swsscommon.Select.TIMEOUT:
+            continue
+        if state != swsscommon.Select.OBJECT:
+            log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+            continue
+
+        (key, op, fvp) = sst.pop()
+        if key in ["PortConfigDone", "PortInitDone"]:
+            break
+
+# Thread wrapper class to update dom info periodically
 class dom_info_update_task:
-    def __init__(self, table):
+    def __init__(self):
+        self.task_thread = None
         self.task_stopping_event = threading.Event()
-        self.task_timer = None
-        self.dom_table = table
 
-    def task_run(self):
-        if self.task_stopping_event.isSet():
-            log_error("Error: dom info update thread received stop event, exiting...")
-            return
+    def task_worker(self):
+        log_info("Start DOM monitoring loop")
 
+        # Connect to STATE_DB and create transceiver dom/sfp info tables
+        state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                         REDIS_HOSTNAME,
+                                         REDIS_PORT,
+                                         REDIS_TIMEOUT_MSECS)
+        int_tbl = swsscommon.Table(state_db, "TRANSCEIVER_INFO")
+        dom_tbl = swsscommon.Table(state_db, "TRANSCEIVER_DOM_SENSOR")
+
+        # Start loop to update dom info in DB periodically
+        while not self.task_stopping_event.wait(DOM_INFO_UPDATE_PERIOD_SECS):
+            logical_port_list = platform_sfputil.logical
+            for logical_port_name in logical_port_list:
+                post_port_dom_info_to_db(logical_port_name, dom_tbl)
+
+        # Delete all the information from DB and then exit
```

Comment:
move this part to the end of the main task?

Reply:
@keboliu done.